hardware_layer/hardware_arduino/src/Motor.cpp

```diff
@@ -32,5 +32,5 @@ void Motor::spin(int pwm)
       digitalWrite(motor_pinA_, LOW);
       digitalWrite(motor_pinB_, LOW);
     }
-    analogWrite(pwm_pin_, abs(pwm));
+    analogWrite(pwm_pin_, 255 - abs(pwm));		 //inverse mapping in the motor driver
```

Comment:
This depends on how the power supply is being given. If that is reversed then you need to do subtract from 255, otherwise, you don't have to. This is fixed at hardware level and we should avoid doing so at the software level to prevent confusion.

Reply:
it would be better if we do it for now since its a basic tendency for all the guys to connect positive to positive and negative to negative. Random things will happen if they fail to do so and then they would take much time to debug the issues.

Reply:
Hm alright